karate-tests/src/test/java/schemas/SensitivePatient.json

```diff
@@ -0,0 +1,43 @@
+{
+    "resourceType": "Patient",
+    "id": "#? utils.isValidNHSNumber(_)",
+    "identifier": [
+      {
+        "system": "https://fhir.nhs.uk/Id/nhs-number",
+        "value": "#? utils.isValidNHSNumber(_)",
+        "extension": [
+          {
+            "url": "https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-NHSNumberVerificationStatus",
+            "valueCodeableConcept": {
+              "coding": [
+                {
+                  "system": "https://fhir.hl7.org.uk/CodeSystem/UKCore-NHSNumberVerificationStatus",
+                  "version": "1.0.0",
+                  "code": "01",
+                  "display": "Number present and verified"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    ],
+    "meta": {
+      "versionId": "#string",
+      "security": [
+        {
+          "system": "http://terminology.hl7.org/CodeSystem/v3-Confidentiality",
+          "code": "#regex(R|V|REDACTED)",
+          "display": "#regex(restricted|very restricted|redacted)"
+        }
+      ]
+    },
+    "gender": "##regex(male|female|other|unknown)",
+    "birthDate": "##? utils.isValidDateString(_)",
+    "deceasedDateTime": "##regex(([12]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])T\\d{2}:\\d{2}:\\d{2}\\+\\d{2}:\\d{2})$|^([12]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])T\\d{2}:\\d{2}:\\d{2})$|^([12]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])T\\d{2}:\\d{2}:\\d{2}\\+\\d{2}:\\d{2})$|^([12]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])))",
```

Comment:
why is regex used for deceasedDateTime and not for birthDate?

Reply:
Good question. That's a very thorough regex too, I've used it in the utils.isValidDateString function now, and both birthDate and deceasedDateTime call upon this function to validate the values